### DISCUSSION TITLE
Timeout for contains value change?

### DISCUSSION THREAD

**Discussion by basickarl** [cypress-io/cypress#17899] [Questions and Help]
Hi I asked a question on Stackoverflow before coming here: https://stackoverflow.com/questions/68937272/timeout-value-for-getallbytestid?noredirect=1#comment121835033_68937272

Is there a way in Cypress to wait for a specific value change in an element? The issue I have is that I have an element (div or whatever) with the value let's say "x", I would like Cypress to grab this element and wait for a maximum of 20 seconds to see if it changes to the value "y". If it doesn't change within the 20 seconds it should timeout.

The following code is the only way I've been able to get it working. The value might change after 1 seconds or 18 seconds, it's random, I'd like the test to run as fast as possible of course (skipping the wait).
```
  cy.wait(20 * 1000);

  cy.getAllByTestId('test-field-one')
    .first()
    .within(() => cy.getByTestId('test-field-two'))
    .should('contain', 'test');
```
Is it possible?

**Comment by bahmutov** [comment]
Please read the intro to Cypress and the retry-ability guide, this is exactly what Cypress has

- https://on.cypress.io/introduction
- https://on.cypress.io/retry-ability
In your case use the command https://on.cypress.io/contains

```js
// retry up to 18 seconds for an element with some selector
// and text "test" inside to appear in the DOM
cy.contains(some selector, 'test', {timeout: 18000})
// if you want to retry until it is in the DOM and is visible to the user, add an assertion
  .should('be.visible')
```
